Make test suite pass with pytest>=7.0

In #202 we added a dev requirement to use pytest<7.0 since the test suite was breaking with the latest pytest version.

Looks like a compatibility issue with `pytest-cases`:
https://github.com/smarie/python-pytest-cases/issues/251

Here's the stack trace:
```
__________________________________________________________________________________________________________________________________________________________________ ERROR collecting tests/test_s3_specific.py __________________________________________________________________________________________________________________________________________________________________
../../miniconda3/envs/cloudpathlib/lib/python3.8/site-packages/pluggy/_hooks.py:265: in __call__
    return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
../../miniconda3/envs/cloudpathlib/lib/python3.8/site-packages/pluggy/_manager.py:80: in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
../../miniconda3/envs/cloudpathlib/lib/python3.8/site-packages/_pytest/python.py:257: in pytest_pycollect_makeitem
    res = list(collector._genfunctions(name, obj))
../../miniconda3/envs/cloudpathlib/lib/python3.8/site-packages/_pytest/python.py:487: in _genfunctions
    fixtures.add_funcarg_pseudo_fixture_def(self, metafunc, fm)
../../miniconda3/envs/cloudpathlib/lib/python3.8/site-packages/_pytest/fixtures.py:164: in add_funcarg_pseudo_fixture_def
    if not metafunc._calls[0].funcargs:
../../miniconda3/envs/cloudpathlib/lib/python3.8/site-packages/pytest_cases/plugin.py:931: in __getitem__
    return self.calls_list[item]
../../miniconda3/envs/cloudpathlib/lib/python3.8/site-packages/pytest_cases/plugin.py:942: in calls_list
    self.create_call_list_from_pending_parametrizations()
../../miniconda3/envs/cloudpathlib/lib/python3.8/site-packages/pytest_cases/plugin.py:994: in create_call_list_from_pending_parametrizations
    remove_empty_ids(callspec)
../../miniconda3/envs/cloudpathlib/lib/python3.8/site-packages/pytest_cases/fixture_parametrize_plus.py:556: in remove_empty_ids
    callspec._idlist = [c for c in callspec._idlist if not c.startswith(EMPTY_ID)]
../../miniconda3/envs/cloudpathlib/lib/python3.8/site-packages/attr/_make.py:626: in _frozen_setattrs
    raise FrozenInstanceError()
E   attr.exceptions.FrozenInstanceError
```

-------------

_Originally posted by @jayqi in https://github.com/drivendataorg/cloudpathlib/pull/202#r800090994_

--- a/cloudpathlib/cloudpath.py
+++ b/cloudpathlib/cloudpath.py
@@ -639,7 +639,6 @@ class CloudPath(metaclass=CloudPathMeta):
         else:
             return path_version
 
-    @property
     def stat(self):
         """Note: for many clients, we may want to override so we don't incur
         network costs since many of these properties are available as

--- a/requirements-dev.txt
+++ b/requirements-dev.txt
@@ -16,7 +16,7 @@ pandas
 pillow
 psutil
 pydantic
-pytest<7.0
+pytest
 pytest-cases
 pytest-cov
 python-dotenv

